internals-js/src/operations.ts

```diff
@@ -2889,13 +3104,13 @@ class InlineFragmentSelection extends FragmentSelection {
         const newSet = new SelectionSetUpdates();
         newSet.add(liftableSelections);
         newSet.add(this.withUpdatedSelectionSet(
-          trimmedSelectionSet.filter((s) => !liftableSelections.includes(s)),
+          normalizedSelectionSet.filter((s) => !liftableSelections.includes(s)),
         ));
         return newSet.toSelectionSet(this.parentType);
       }
     }
 
-    return this.selectionSet === trimmedSelectionSet ? this : this.withUpdatedSelectionSet(trimmedSelectionSet);
+    return this.selectionSet === normalizedSelectionSet ? this : this.withUpdatedSelectionSet(normalizedSelectionSet);
```

Comment:
I'm noticing that unlike `FieldSelection.normalize()` and `InlineFragmentSelection.normalize()` which update `this.parentType` to `parentType`, `InlineFragmentSelection.normalize()` doesn't seem to, e.g.:
- The `return this.selectionSet === normalizedSelectionSet ? this : this.withUpdatedSelectionSet(normalizedSelectionSet);` doesn't update the element's `parentType`.
- The `return this.withUpdatedSelectionSet(selectionSetOfElement( ... )` above doesn't update the element's `parentType`.
- The `return newSet.toSelectionSet(this.parentType)` above uses `this.parentType` instead of `parentType`.
- There's a `(this.element.typeCondition ?? this.parentType).typenameField()!` call above instead of `(thisCondition ?? parentType).typenameField()!,`

Also I noticed that when you lift selections out of the fragment in this function, the `parentType` isn't accordingly updated for those selections.

Is this fine/expected? (It might be that the field is only used in certain cases, or maybe some other code requires it to not be changed, but I'm not familiar enough with the code here.)

Reply:
No,I was being sloppy.

I mean, it is probably mostly fine in the grand scheme of stuff because when we build a `SelectionSet` with `SelectionSetUpdates` (which `SelectionSet.normalize` does through `lazyMap`), then it ensures the elements have the proper parent type by [calling `rebasOn`](https://github.com/apollographql/federation/blob/main/internals-js/src/operations.ts#L1996C1-L2022), so I believe the selections still ended up with the proper parent type ultimately.

Still, that was sloppy and would be unexpected at least out of the `Selection.normlize` call and could cause issue someday, so fixed it.

> you lift selections out of the fragment in this function, the `parentType` isn't accordingly updated for those selections

As mentioned above, the fact that we passed the lifted selections to a `SelectionSetUpdates` took care of updating the parent type of the elements (of course, as you pointed out, the `toSelectionSet` call for that `SelectionSetUpdates` was incorrect, so did fixed that, but just calling out that there is nothing additional to do).